sns_ik_lib/matlab/snsIk_acc_QP_mt.m

```diff
@@ -0,0 +1,130 @@
+function [ddq, sData, exitCode] = snsIk_acc_QP_mt(ddqLow, ddqUpp, ddxGoalData, JData, dJdqData, alpha)
+% [ddq, sData, exitCode] = snsIk_vel_QP(ddqLow, ddqUpp, ddxGoalData, JData, dJdqData, alpha)
+%
+% This function implements the generic multi-task version of SNS-IK acceleration
+% solver using QP.
+%
+% INPUTS:
+%   ddqLow: lower limit for JDataoint accelerations
+%   ddqUpp: upper limit for JDataoint accelerations
+%   ddxGoalDataData: task-space accelerations of all the tasks in the order of priorities
+%   JData: Jacobians of all the task in corresponding order
+%   dJdqData: dJ * dq of all the task in corresponding order
+%   alpha: trade-off between minimum-joint velocity and maximum task scale
+%      default: alpha = 1e-3
+%      small alpha: favor solutions that maximize task scale
+%      large alpha: favor minimum-joint velocity solutions
+%
+% OUTPUTS:
+%   ddq = Joint acceleration solution with maximum task scale factor
+%   sData = task scale factors [0, 1] for all the tasks
+%   exitCode    (1 == success)
+%
+% NOTES:
+
+% Licensed under the Apache License, Version 2.0 (the "License");
+% you may not use this file except in compliance with the License.
+% You may obtain a copy of the License at
+% http://www.apache.org/licenses/LICENSE-2.0
+%
+% Unless required by applicable law or agreed to in writing, software
+% distributed under the License is distributed on an "AS IS" BASIS,
+% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+% See the License for the specific language governing permissions and
+% limitations under the License.
+
+% TODO: input validation
+% TODO: return optimization status
+
+if nargin < 5
+   alpha = 1e-3;
+end
+
+% initialize variables
+numTask = size(JData,1);
+nJnt = numel(ddqLow);
+ddqData = cell(numTask,1);
+sData = zeros(numTask,1);
+resData = cell(numTask,1);
+tol = 1e-6;
+
+% initialize problem
+problem.options = optimset('Display','off');
+problem.solver = 'quadprog';
+
+for iTask = 1:numTask
```

Comment:
Is there a way to solve all tasks with a single QP solve? I see two possible reasons for this:
1) It might find a better solution than solving it sequentially
2) It might be faster to call the solver once.

Definitely keep this implementation - it is great. I'm more thinking as a possible additional version of the algorithm.

Reply:
I need to think a bit more about this. It is not straightforward since some of the tasks are infeasible, I don't know how the QP will figure the feasibility of each task when it is formulated in a big QP problem. I will leave this in TODO note and we will create another QP example when we figure it out.

Reply:
That sounds good - thanks!